### PR TITLE
WP 137 - integration tests to start server and serve routes

### PR DESCRIPTION
**Pull request by mmcgahan**
So far, none of our unit tests actually start the platform server - this PR adds some 'component'/'integration' tests to the server start function to make sure a dummy app can start and serve routes.

Big change: new `tests_integration/` root-level directory. Since integration/component tests aren't tied to a single module, it would be awkward to try to keep them alongside the modules they integrate. Instead, I've created a new directory and a new naming pattern to distinguish unit tests (which are run by `npm run test:unit` and include code coverage) and integration tests (which are run by `npm run test:integration` and do not generate a coverage report).

`npm test` now runs _both_ types of tests in sequence - unit testing first, then integration tests.

### CODE REVIEW

File: jest.int.config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "setupFiles": [
+    "<rootDir>/src/util/globals.js"
+  ],
+  "moduleFileExtensions": [
+    "js",
+    "jsx"
```
Comment (mattkime) [inline jest.int.config.json]: will integration tests ever be `.jsx` files?

**Comment by mattkime** [comment]
lgtm

File: jest.int.config.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  ],
+  "testRegex": "tests_integration\/.*\\.int\\.jsx?$"
```
Comment (sadafie) [inline jest.int.config.json]: if we're ever going to have other types of tests in a test directory, i'd opt for `tests/integration/` here instead.

Comment (mmcgahan) [inline jest.int.config.json]: i don't _think_ we would ever have other types of tests, but `tests/integration/` works for me! Will update and merge unless you wanted to look at anything else

**Pull request merged**